core/src/services/huggingface/message.rs

```diff
@@ -0,0 +1,401 @@
+                        "id": "bc1ef030bf3743290d5e190695ab94582e51ae2f",
+                        "title": "Upload 141 files",
+                        "date": "2023-11-17T23:50:28.000Z"
+                    },
+                    "security": {
+                        "blobId": "45fa7c3d85ee7dd4139adbc056da25ae136a65f2",
+                        "name": "maelstrom/lib/maelstrom.jar",
+                        "safe": true,
+                        "avScan": {
+                            "virusFound": false,
+                            "virusNames": null
+                        },
+                        "pickleImportScan": {
+                            "highestSafetyLevel": "innocuous",
+                            "imports": [
+                                {"module": "torch", "name": "FloatStorage", "safety": "innocuous"},
+                                {"module": "collections", "name": "OrderedDict", "safety": "innocuous"},
+                                {"module": "torch", "name": "LongStorage", "safety": "innocuous"},
+                                {"module": "torch._utils", "name": "_rebuild_tensor_v2", "safety": "innocuous"}
+                            ]
+                        }
+                    }
+                }
+            ]
+            "#,
+        );
+
+        let decoded_response = serde_json::from_slice::<Vec<HuggingFaceStatus>>(&resp)
+            .map_err(new_json_deserialize_error)?;
+
+        assert_eq!(decoded_response.len(), 1);
+
+        assert_eq!(decoded_response[0].type_, "file");
+        assert_eq!(
+            decoded_response[0].oid,
+            "45fa7c3d85ee7dd4139adbc056da25ae136a65f2"
+        );
+        assert_eq!(decoded_response[0].size, 69512435);
+        assert_eq!(decoded_response[0].path, "maelstrom/lib/maelstrom.jar");
+
+        assert_eq!(
+            decoded_response[0].lfs.as_ref().unwrap().oid,
+            "b43f4c2ea569da1d66ca74e26ca8ea4430dfc29195e97144b2d0b4f3f6cafa1c"
+        );
+
+        assert_eq!(decoded_response[0].lfs.as_ref().unwrap().size, 69512435);
+        assert_eq!(decoded_response[0].lfs.as_ref().unwrap().pointer_size, 133);
+
+        assert_eq!(
+            decoded_response[0].last_commit.as_ref().unwrap().id,
+            "bc1ef030bf3743290d5e190695ab94582e51ae2f"
+        );
+        assert_eq!(
+            decoded_response[0].last_commit.as_ref().unwrap().title,
+            "Upload 141 files"
+        );
+        assert_eq!(
+            decoded_response[0].last_commit.as_ref().unwrap().date,
+            "2023-11-17T23:50:28.000Z"
+        );
+
+        assert_eq!(
+            decoded_response[0].security.as_ref().unwrap().blob_id,
+            "45fa7c3d85ee7dd4139adbc056da25ae136a65f2"
+        );
+        assert_eq!(
+            decoded_response[0].security.as_ref().unwrap().name,
+            "maelstrom/lib/maelstrom.jar"
+        );
+        assert_eq!(decoded_response[0].security.as_ref().unwrap().safe, true);
+
+        assert_eq!(
```

Comment:
We can implement `Eq` for those types, so we can `assert_eq!(decoded_response[0], TypeAbc)` directly.

Reply:
Done